### PR TITLE
PIA-909: Server Connection Tests

### PR DESCRIPTION
**Pull request by xv-geneva-parayno**
Added ServerConnectionTests

### CODE REVIEW

**Review by kp-said-rehouni — COMMENTED** [review]
Great work Geneva!
Just one thing, it seems that some of the `ServerConnectionTests` tests are failing: https://github.com/pia-foss/mobile-ios/actions/runs/6978486538/job/18989975797?pr=41

File: PIA-VPN_E2E_Tests/Screens/HomeScreen.swift [hunk 1]
```diff
@@ -117,4 +125,16 @@ extension XCUIApplication {
...
+        if (disconnectedStatusLabel.exists) {
+            connectionButton.tap()
+        }
+    }
+    
+    func disconnectToVPN() {
+        if (connectedStatusLabel.exists) {
+            connectionButton.tap()
+        }
+    }
```
Comment (kp-laura-sempere) [inline PIA-VPN_E2E_Tests/Screens/HomeScreen.swift:139]: Nice!

File: PIA-VPN_E2E_Tests/Tests/ServerConnectionTests.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    app.disconnectToVPN()
+                    app.navigateToRegionSelection()
+                    app.searchRegion(regionName: "Philippines").firstMatch.tap()
+                    expect(app.connectedStatusLabel.exists).to(beTrue())
+                    expect(app.vpnServerButton.staticTexts["Philippines"].exists).to(beTrue())
+                }
+            }
+        }
+    }
+}
```
Comment (kp-laura-sempere) [inline PIA-VPN_E2E_Tests/Tests/ServerConnectionTests.swift:45]: These tests look great!

When they run on a simulator, they are using a mock connection which fakes the connection and disconnection without any delay. This is great to verify quickly the UI updates when we toggle the connection.

Should we also maybe create another test suite that covers the same but runs on a real device and hence using a real connection? We would not run those in CI, only manually on our local machines. Let me know your thoughts and maybe we can create a separate ticket to add a new test plan for real devices that don't run on CI.
But nevertheless, this test suite is perfect as is on CI to verify the UI when toggling the connection and select a different region 👍

**Review by kp-laura-sempere — APPROVED** [review]
LGTM! 🚀 

I just added a small suggestion to address on a separate ticket.

**Pull request merged**